doc/decisions/search_plants.md

```diff
@@ -90,11 +93,25 @@ Cons:
 
 ## Decision
 
+Given the constrains and assumptions Partial Match with String Similarity (pg_trgm) seems to be the best option for us.
+
 ## Rationale
 
+As there is a relatively small number of rows and search terms performance isn't that much of an issue.
+
+The ability to handle spelling mistakes and loose matches is also great for us as plant names can be somewhat complicated to spell.
+
+Generally it is seems to fit best for our usecase.
+Something like elasticsearch could cover the usecase better, however it adds a lot of complexity.
+
 ## Implications
 
--
+The following points are difficult if not impossible to implement using PostgreSQL.  
+Before trying them it should be evaluated if a different database like elasticsearch might be a better fit for this usecase.
+
+- Other columns can be matched with extra syntax (e.g. environmental fit or ecological value)
```

Comment:
Can you elaborate? We were thinking about something like `water_requirement = dry`, which basically simply adds one WHERE clause. Especially if the SQL query is text already, it should be simple to implement (if you keep the extra syntax simple).

Reply:
Apart from what I wrote here I can tell you from experience that knowbody would ever use a feature like that apart from the developers/designer themselves.
I have seen exactly that with use case with elasticsearch 2 times at different workplaces and in both cases no user actually used that feature (even though it was natively supported).
The reason being is that the user would have to know all column names in the database by heart or at least look at the documentation an figure out which variants to enter where.
A solution that I have commonly seen for that is to provide additional fields in the frontend where the user can enter e.g. water_requirement if they want to. However then there is no need to parse the query in the backend as it could simply be part of the query struct (although that might be difficult as well in the current situation as again we have to be careful about SQL injections).

Reply:
Yes, I agree. Let us remove this "special syntax" feature completely (you can remove it from the use case). At some later time (not this term) we implement a proper extended search.